optaplanner-core/src/main/java/org/optaplanner/core/impl/heuristic/selector/move/factory/MoveIteratorFactory.java

```diff
@@ -49,7 +50,7 @@
      * @throws UnsupportedOperationException if only {@link #createRandomMoveIterator(ScoreDirector, Random)} is
      * supported
      */
-    Iterator<Move> createOriginalMoveIterator(ScoreDirector<Solution_> scoreDirector);
+    Iterator<Move_> createOriginalMoveIterator(ScoreDirector<Solution_> scoreDirector);
```

Comment:
Thanks Jiri.
It's a nice experiment, but the extra generic type isn't worth it's weight in complexity (and it also breaks backw. compatbility on the side although it's not public api so it doesn't factor into the decision much).
+1 to remove commit 3 & 4